kernel/src/main/scala/org/apache/toree/kernel/api/Kernel.scala

```diff
@@ -396,7 +372,29 @@ class Kernel (
     interpreterManager.interpreters.get(name)
   }
 
-  override def sparkSession: SparkSession = SparkSession.builder.getOrCreate
+  private lazy val defaultSparkConf: SparkConf = createSparkConf(
+    new SparkConf().setAppName(SparkKernelInfo.banner))
```

Comment:
I patched my fork with this PR and it looks good - although I did confirm it reverts the change in PR 107.  That said, I also confirmed that removing `.setAppName(SparkKernelInfo.banner)` restores the expected behavior and allows for --name overrides.